docs/compatibility.md

```diff
@@ -89,3 +89,7 @@ If you need to mock HTTPX the same way that test utilities like `responses` and
 `requests` defers most of its HTTP networking code to the excellent [`urllib3` library](https://urllib3.readthedocs.io/en/latest/).
 
 On the other hand, HTTPX uses [HTTPCore](https://github.com/encode/httpcore) as its core HTTP networking layer, which is a different project than `urllib3`.
+
+## Query Parameters
+
+`requests` omits `params` whose values are `None` (e.g. `requests.get(..., params={"foo": None})`). HTTPX will include a query parameter without a value instead. 
```

Comment:
I'd probably(?) rather we were raising a TypeError here, since our annotations don't include None as a valid value for items in params. It's only by accident of implementation that it happens to currently end up being treated as equivalent to `""`.

Reply:
Makes sense — still, we can add these docs, probably just tweaking them like this:

```suggestion
`requests` omits `params` whose values are `None` (e.g. `requests.get(..., params={"foo": None})`). This is not supported by HTTPX.
```